src/server/conn/auto.rs

```diff
@@ -87,6 +88,36 @@ impl<E> Builder<E> {
         Ok(())
     }
 
+    /// Bind a connection together with a [`tower::Service`].
+    pub async fn serve_connection_tower<I, S, B>(&self, io: I, service: S) -> Result<()>
+    where
+        S: tower::Service<Request<Incoming>, Response = Response<B>> + Clone + Send,
+        S::Future: 'static,
+        S::Error: Into<Box<dyn StdError + Send + Sync>>,
+        B: Body + Send + 'static,
+        B::Data: Send,
+        B::Error: Into<Box<dyn StdError + Send + Sync>>,
+        I: AsyncRead + AsyncWrite + Unpin + 'static,
+        E: Http2ConnExec<TowerToHyperServiceFuture<S, Request<Incoming>>, B>,
+    {
+        let (version, io) = read_version(io).await?;
+        let io = TokioIo::new(io);
+        match version {
+            Version::H1 => {
+                self.http1
+                    .serve_connection(io, TowerToHyperService::new(service))
+                    .await?
+            }
+            Version::H2 => {
+                self.http2
+                    .serve_connection(io, TowerToHyperService::new(service))
+                    .await?
+            }
+        }
+
+        Ok(())
```

Comment:
Ideally we'd just call `serve_connection` but that requires `S: 'static`. Not totally sure why that is. Copy-pasting the impl doesn't require that 🤔